crates/polars-io/src/json/mod.rs

```diff
@@ -287,6 +287,8 @@ where
                     }
                 }
 
+                let allow_extra_fields_in_struct = self.schema.is_some();
```

Comment:
We currently have this behavior where a user can perform a column selection within a struct field by giving a partial schema - this happens because of the existing behavior of JSON deserialization to silently ignore extra fields.

In this PR we change that behavior to raise an error, but to maintain the ability to do this column selection I made it so that we don't raise when we see extra keys if the schema was provided by the user.